src/utils/DatasetUtils.js

```diff
@@ -0,0 +1,27 @@
+import {find, getOr, isNil} from 'lodash/fp';
+import {isEmpty} from 'lodash/fp';
+import {DataUseTranslation} from '../libs/dataUseTranslation';
+
+export const findPropertyValue = (dataSet, propName, defaultVal) => {
+  const defaultValue = isNil(defaultVal) ? '' : defaultVal;
+  return getOr(defaultValue)('propertyValue')(find({ propertyName: propName })(dataSet.properties));
+};
+
+export const getDataUseCodes = async (dataset) => {
```

Comment:
I think this should return the data use codes instead of modifying dataset. I think it would confuse me / would worry me about side effects if I was another developer and was using this function. this could be kept as react state on the component level and sent down to everything that needs it.